site/content/entry/yuzu-progress-report-may-2022/index.md

```diff
@@ -0,0 +1,272 @@
+
+For those souls that happen to land in this situation, a button labeled "Check for Working Vulkan" at the bottom of the Graphics settings window will show up, allowing to retest Vulkan support.
+
+{{< imgs
+    "./button.png| Once you manage to solve the issue, click on the button at the bottom!"
+  >}}
+
+Thanks to [toastUnlimited](https://github.com/lat9nq), gone is OpenGL as the default graphics API. 
+Out with the old, in with the new. {{< gh-hovercard "8393" "Long live King Vulkan." >}}
+
+Going forward, our developers will focus their priority to Vulkan, but will continue to support OpenGL.
+OpenGL users are recommended to use the GLSL shader backend, as GLASM and SPIR-V will receive limited support from now on.
+
+## Graphical changes, driver issues, and the nostalgia bliss that is the good old 64
+
+This month, [byte[]](https://github.com/liamwhite) continues the wave of improvements for `Super Mario 3D All-Stars`, this time, he noticed a bug in the DMAcopy ([direct memory access](https://en.wikipedia.org/wiki/Direct_memory_access)) of the Nintendo Switch’s GPU.
+
+`DMACopy` is a mechanism that many games use to send texture data to the GPU, it handles the format conversion from "pitch" (pixels on a line by line basis) to "tiled" (gridded) images.
+This process works by writing the pitch image data into GPU memory accessible by the DMA engine. Next, a DMAcopy is requested through the DMA engine driver, converting the image data into a separate buffer accessible by the GPU. This buffer will then be used as the texture on the final draw.
+
+{{< gh-hovercard "8313" "After fixing `bytes_per_pixel`," >}} `Super Mario Galaxy` now has proper lens flare.
+
+{{< single-title-imgs-compare
+    "RTX On? (Super Mario Galaxy)"
+    "./dmabug.png"
+    "./dmafix.png"
+>}}
+
+byte[] also improved the way OpenGL interprets face flips depth, [replacing the previously reported fix](https://yuzu-emu.org/entry/yuzu-progress-report-apr-2022/#saving-princess-peach-yet-again). The previous implementation had terrible peformance in OpenGL, making it virtually unplayable.
+
+While this wasn't an issue while using Vulkan, now `Super Mario 64` and `Super Mario Galaxy` {{< gh-hovercard "8314" "are playable in both APIs." >}} 
+Fermi GPU users rejoyce.
+
+{{< imgs
+    "./sm64.png| It's-a Mario, now running fast in OpenGL! (Super Mario 64)"
+  >}}
+
+One of the important parts of yuzu's graphical emulation is the need to translate small sets of GPU instructions, called `macros`. 
+yuzu uses a Just-in-Time (JIT) compiler to emulate these macros in a performant way.
+MacroJIT is an optimization applied over this process, it manages to provide a performance boost of plus 10%, more or less.
+
+byte[] found that macro could try to access a parameter that was too far outside the bounds of what it was supposed to be accessing. 
+Usually the end result of this behaviour would be the emulator crashing.
+{{< gh-hovercard "8319" "One less reason for annoying crashes." >}}
```

Comment:
```suggestion
byte[] found that due to emulation inaccuracies, sometimes a macro could try to access a parameter that was too far outside the bounds of what it was supposed to be accessing. This could crash the emulator without a single trace as to why in some cases.
{{< gh-hovercard "8319" "One less reason for annoying crashes." >}}
```